File: DSA/String/Code/InsertACharacter.java
// Insert a Character in a String in Java
// Insert character at specific position

public class InsertACharacter {
    static String insertChar(String str, int position, char ch) {
        return str.substring(0, position) + ch + str.substring(position);
    }
    
    public static void main(String[] args) {
        String str = "Hello World";
        
        System.out.println("Original string: " + str);
        
        // Method 1: Using substring and concatenation
        System.out.println("\nUsing substring:");
        String result1 = str.substring(0, 0) + 'H' + str.substring(0);
        System.out.println("After inserting 'H' at position 0: " + result1);
        
        // Method 2: Insert at position 5
        String result2 = str.substring(0, 5) + '!' + str.substring(5);
        System.out.println("After inserting '!' at position 5: " + result2);
        
        // Method 3: Insert at end
        String result3 = str + '!';
        System.out.println("After inserting '!' at end: " + result3);
        
        // Method 4: Using StringBuilder
        System.out.println("\nUsing StringBuilder:");
        StringBuilder sb = new StringBuilder(str);
        sb.insert(0, 'H');
        System.out.println("After inserting 'H' at position 0: " + sb.toString());
        
        // Method 5: Insert at multiple positions
        StringBuilder sb2 = new StringBuilder(str);
        sb2.insert(0, '*');
        sb2.insert(sb2.length(), '*');
        System.out.println("After inserting '*' at both ends: " + sb2.toString());
        
        // Method 6: Insert string of characters
        StringBuilder sb3 = new StringBuilder(str);
        sb3.insert(5, "***");
        System.out.println("After inserting '***' at position 5: " + sb3.toString());
        
        // Method 7: Using function
        System.out.println("\nUsing function:");
        String result4 = insertChar(str, 6, '@');
        System.out.println("After inserting '@' at position 6: " + result4);
    }
}

File: DSA/String/Code/InsertACharacter.cs
using System;
using System.Text;

// Insert a Character in a String in C#
// Insert character at specific position

class InsertACharacter {
    static string InsertChar(string str, int position, char ch) {
        return str.Insert(position, ch.ToString());
    }
    
    static void Main() {
        string str = "Hello World";
        
        Console.WriteLine("Original string: " + str);
        
        // Method 1: Using Insert()
        Console.WriteLine("\nUsing Insert():");
        string result1 = str.Insert(0, "H");
        Console.WriteLine("After inserting 'H' at position 0: " + result1);
        
        // Method 2: Insert at position 5
        string result2 = str.Insert(5, "!");
        Console.WriteLine("After inserting '!' at position 5: " + result2);
        
        // Method 3: Insert at end
        string result3 = str.Insert(str.Length, "!");
        Console.WriteLine("After inserting '!' at end: " + result3);
        
        // Method 4: Using Substring
        Console.WriteLine("\nUsing Substring:");
        string result4 = str.Substring(0, 5) + "*" + str.Substring(5);
        Console.WriteLine("After inserting '*' at position 5: " + result4);
        
        // Method 5: Using StringBuilder
        Console.WriteLine("\nUsing StringBuilder:");
        StringBuilder sb = new StringBuilder(str);
        sb.Insert(0, '*');
        Console.WriteLine("After inserting '*' at position 0: " + sb.ToString());
        
        // Method 6: Multiple insertions
        StringBuilder sb2 = new StringBuilder(str);
        sb2.Insert(0, '*');
        sb2.Insert(sb2.Length, '*');
        Console.WriteLine("After inserting '*' at both ends: " + sb2.ToString());
        
        // Method 7: Using function
        Console.WriteLine("\nUsing function:");
        string result5 = InsertChar(str, 6, '@');
        Console.WriteLine("After inserting '@' at position 6: " + result5);
    }
}
